src/helpers/geoLocation.ts

```diff
@@ -0,0 +1,54 @@
+import cloneDeep from 'lodash/cloneDeep';
+import isEmpty from 'lodash/isEmpty';
+import set from 'set-value';
+import { FixMe } from '../util/types';
+import { getKeyAndValueFromMessage } from '../v0/util';
+import * as GenericFieldMappingJson from '../v0/util/data/GenericFieldMapping.json';
+
+export default class GeoLocationHelper {
+  public static getAddressKeyAndValue(message: Record<string, FixMe>): {
+    addressKey: string;
+    address: Record<string, FixMe>;
+  } {
+    const { value: address, key: foundKey } =
+      getKeyAndValueFromMessage(message, GenericFieldMappingJson.address) || {};
+    const { key: traitsKey } = getKeyAndValueFromMessage(message, GenericFieldMappingJson.traits);
+    let addressKey = foundKey;
+    if (!foundKey) {
+      addressKey = `${traitsKey}.address`;
+      if (!traitsKey) {
+        [addressKey] = GenericFieldMappingJson.address;
+      }
+    }
+    return { addressKey, address };
+  }
+
+  public static getGeoLocationData(message: Record<string, FixMe>): Record<string, FixMe> {
```

Comment:
The function name seems like a misnomer, it's not gettingGeoLocationData, it's enriching message with geoData right ? Can we fix the naming as well if that's the case ?

Reply:
Will look at some better name for this

Reply:
Tried with a different name. Please check and let me know your thoughts